msg-socket/src/pub/socket.rs

```diff
@@ -61,22 +66,24 @@ impl<T: ServerTransport> PubSocket<T> {
     }
 
     /// Binds the socket to the given address. This spawns the socket driver task.
-    pub async fn bind(&mut self, addr: &str) -> Result<(), PubError> {
-        // Take the transport here, so we can move it into the backend task
-        let mut transport = self.transport.take().unwrap();
-
+    pub async fn bind(&mut self, addr: SocketAddr) -> Result<(), PubError> {
```

Comment:
nit: what do you think of following the tokio convention of using the generic `ToSocketAddrs` trait?
It's the one used in TcpStream::connect:

```rust
pub async fn connect<A: ToSocketAddrs>(addr: A) -> io::Result<TcpStream> {
    let addrs = to_socket_addrs(addr).await?;

    let mut last_err = None;

    for addr in addrs {
        match TcpStream::connect_addr(addr).await {
            Ok(stream) => return Ok(stream),
            Err(e) => last_err = Some(e),
        }
    }

   // ...
```

the cool thing is that it automatically resolves dns names which is something we need to do manually right now

Reply:
Was also thinking about this, will make issue w.r.t. QoL improvements